extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeSignatureUtils.java

```diff
@@ -0,0 +1,150 @@
+package org.schabi.newpipe.extractor.services.youtube;
+
+import org.schabi.newpipe.extractor.exceptions.ParsingException;
+import org.schabi.newpipe.extractor.utils.JavaScript;
+import org.schabi.newpipe.extractor.utils.Parser;
+import org.schabi.newpipe.extractor.utils.jsextractor.JavaScriptExtractor;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Utility class to get the signature timestamp of YouTube's base JavaScript player and deobfuscate
+ * signature of streaming URLs from HTML5 clients.
+ */
+final class YoutubeSignatureUtils {
+
+    /**
+     * The name of the deobfuscation function which needs to be called inside the deobfuscation
+     * code.
+     */
+    static final String DEOBFUSCATION_FUNCTION_NAME = "deobfuscate";
+
+    private static final String[] FUNCTION_REGEXES = {
+            "\\bm=([a-zA-Z0-9$]{2,})\\(decodeURIComponent\\(h\\.s\\)\\)",
+            "\\bc&&\\(c=([a-zA-Z0-9$]{2,})\\(decodeURIComponent\\(c\\)\\)",
+            // CHECKSTYLE:OFF
+            "(?:\\b|[^a-zA-Z0-9$])([a-zA-Z0-9$]{2,})\\s*=\\s*function\\(\\s*a\\s*\\)\\s*\\{\\s*a\\s*=\\s*a\\.split\\(\\s*\"\"\\s*\\)",
+            // CHECKSTYLE:ON
+            "([\\w$]+)\\s*=\\s*function\\((\\w+)\\)\\{\\s*\\2=\\s*\\2\\.split\\(\"\"\\)\\s*;"
+    };
+
+    private static final String STS_REGEX = "signatureTimestamp[=:](\\d+)";
+
+    private static final String DEOBF_FUNC_REGEX_START = "(";
+    private static final String DEOBF_FUNC_REGEX_END = "=function\\([a-zA-Z0-9_]+\\)\\{.+?\\})";
+
+    private static final String SIG_DEOBF_HELPER_OBJ_NAME_REGEX = ";([A-Za-z0-9_\\$]{2,})\\...\\(";
+    private static final String SIG_DEOBF_HELPER_OBJ_REGEX_START = "(var ";
+    private static final String SIG_DEOBF_HELPER_OBJ_REGEX_END = "=\\{(?>.|\\n)+?\\}\\};)";
```

Comment:
Why don't these need `@SuppressWarnings("RegExpRedundantEscape")`? They also use an escaped `}`

Reply:
My IDE didn't complain about that, and I didn't disable the corresponding warning. Is a warning shown for you or someone else?

Reply:
Yeah don't worry then, it was just to make sure